pkg/repositories/database.go

```diff
@@ -94,14 +104,16 @@ func GetDB(ctx context.Context, dbConfig *runtimeInterfaces.DbConfig, logConfig
 			ExtraOptions: dbConfig.DeprecatedExtraOptions,
 			Debug:        dbConfig.DeprecatedDebug,
 		}
-		dialector = postgres.Open(getPostgresDsn(ctx, pgConfig))
+
 	default:
 		panic(fmt.Sprintf("Unrecognized database config %v", dbConfig))
 	}
-	gormDb, err := gorm.Open(dialector, &gorm.Config{
+
+	gormConfig := &gorm.Config{
 		Logger:                                   gormLogger.Default.LogMode(logLevel),
 		DisableForeignKeyConstraintWhenMigrating: !dbConfig.EnableForeignKeyConstraintWhenMigrating,
-	})
+	}
+	gormDb, err := createIfNotExists(ctx, pgConfig, gormConfig)
```

Comment:
please also move this to postgres-specific case statements

Reply:
Done.